nestkernel/ntree_impl.h

```diff
@@ -391,7 +393,11 @@ Ntree< D, T, max_capacity, max_depth >::subquad_( const Position< D >& pos )
   int r = 0;
   for ( int i = 0; i < D; ++i )
   {
-    r += ( 1 << i ) * ( pos[ i ] < lower_left_[ i ] + extent_[ i ] / 2 ? 0 : 1 );
+    // Comparing against an epsilon value in case there are round-off errors.
+    // Using a negative epsilon value because the round-off error may go both ways
+    // and the difference we check against may therefore be +/- 10^-16.
+    r += ( 1 << i )
+      * ( ( ( lower_left_[ i ] + extent_[ i ] / 2 ) - pos[ i ] ) > -std::numeric_limits< double >::epsilon() ? 0 : 1 );
```

Comment:
To make this more readable, I'd suggest 
```suggestion
    const bool in_left_half = ( ( lower_left_[ i ] + extent_[ i ] / 2 ) - pos[ i ] ) > -std::numeric_limits< double >::epsilon();
    r += ( 1 << i ) * ( in_left_half ? 0 : 1 );
```

I think as a temporary fix this is fine, but we should revisit the issue again later. My main concern is that this test will work as long as coordinates are close to 1, but of they are much smaller or larger, we need to consider relative error. I would feel more comfortable with an approach where we explicitly pad boundaries in some consistent way.